guides/03_building-with-blocks/06_theming-guide

```diff
@@ -0,0 +1,177 @@
+
+## Extending Themes via the Constructor
+
+Although each theme has hundreds of CSS variables, the values for most these variables are drawn from 8 core variables which can be set through the constructor of each prebuilt theme. Modifying these 8 arguments allows you to quickly change the look and feel of your app.
+
+### Core Colors
+
+The first 3 constructor arguments set the colors of the theme and are gradio.themes.Color objects. Internally, these Color objects hold brightness values for the palette of a single hue, ranging from 50, 100, 200..., 800, 900, 950. Other CSS variables are derived from these 3 colors.
+
+The 3 color constructor arguments are:
+- `primary_hue`: This is the color draws attention in your theme. In the default theme, this is set to gradio.themes.Orange.
+- `secondary_hue`: This is the color that is used for secondary elements in your theme. In the default theme, this is set to gradio.themes.Blue.
+- `neutral_hue`: This is the color that is used for text and other neutral elements in your theme. In the default theme, this is set to gradio.themes.Gray.
+
+You could modify these values using their string shortcuts, such as
+
+```python
+with gr.Blocks(theme=gr.themes.Default(primary_hue="red", secondary_hue="pink")):
+    ...
+```
+
+or you could use the Color object directly, such as
+
+```python
+with gr.Blocks(theme=gr.themes.Default(primary_hue=gr.themes.Red, secondary_hue=gr.themes.Pink)):
+    ...
+```
+You could also create your own custom Color objects and pass them in.
+
+### Core Sizing
+
+The next 3 constructor arguments set the sizing of the theme and are gradio.themes.Size objects. Internally, these Size objects hold pixel size values that range from `xxs` to `xxl`. Other CSS variables are derived from these 3 sizes.
+
+- `spacing_size`: This sets the padding within and spacing between elements. In the default theme, this is set to gradio.themes.spacing_md.
+- `radius_size`: This sets the roundedness of corners of elements. In the default theme, this is set to gradio.themes.radius_md.
+- `text_size`: This sets the font size of text. In the default theme, this is set to gradio.themes.font_md.
+
+You could modify these values using their string shortcuts, such as
+
+```python
+with gr.Blocks(theme=gr.themes.Default(spacing_size="sm", radius_size="lg")):
+    ...
+```
+
+or you could use the Size object directly, such as
+
+```python
+with gr.Blocks(theme=gr.themes.Default(spacing_size=gr.themes.spacing_sm, radius_size=gr.themes.radius_lg)):
+    ...
+```
+
+You could also create your own custom Size objects and pass them in.
+
+### Core Fonts
+
+The final 2 constructor arguments set the fonts of the theme. You can pass a list of fonts to each of these arguments to specify fallbacks. If you provide a string, it will be loaded as a system font. If you provide a gradio.themes.GoogleFont, the font will be loaded from Google Fonts.
```

Comment:
```suggestion
The final 2 constructor arguments set the fonts of the theme. You can pass a list of fonts to each of these arguments to specify fallbacks. If you provide a string, it will be loaded as a system font. If you provide a `gradio.themes.GoogleFont`, the font will be loaded from Google Fonts.
```